Sources/Mocker/Mock.swift

```diff
@@ -125,12 +137,38 @@ public struct Mock: Equatable {
     /// Creates a `Mock` for the given data type. The mock will be automatically matched based on a URL created from the given parameters.
     ///
     /// - Parameters:
-    ///   - dataType: The type of the data which is returned.
+    ///   - dataType: The type of the data which designates the Content-Type header.
     ///   - statusCode: The HTTP status code to return with the response.
     ///   - data: The data which will be returned as the response based on the HTTP Method.
     ///   - additionalHeaders: Additional headers to be added to the response.
+    @available(*, deprecated, message: "Use init(contentType: DataType?, ...) instead.")
```

Comment:
We can use `renamed`, which although a bit awkward as it is not "type safe", creates an auto fix-it in Xcode,
```suggestion
    @available(*, deprecated, renamed: "init(contentType:statusCode:data:additionalHeaders:)")
```

Reply:
Two things I've just learned: This `renamed` stuff. And that GitHub reviews can have code suggestions! Both are cool features, thanks!